File: control-assurance-sp/src/types/GoElementEvidence.ts
import { IEntity } from "./Entity";
import { User } from "./User";

export interface IGoElementEvidence  extends IEntity {
    IsLink?:boolean;
    Details?: string;
    Controls?: string;
    Team?: string;
    InfoHolder?: string;
    AdditionalNotes?: string;
    GoElementId?: number; 
    DateUploaded?: Date;
    UploadedByUserId?: number;
    //User?: User;

}

export class GoElementEvidence implements IGoElementEvidence{ 
    public ID: number = 0;
    public Title: string = null;
    public Details?: string = "";
    public Controls?: string = "";
    public Team?: string = "";
    public InfoHolder?: string = "";
    public AdditionalNotes?: string = "";
    public GoElementId?: number = null;
    public DateUploaded?: Date = null;
    public UploadedByUserId?: number = null;

    constructor(goElementId: number) {
        this.GoElementId = goElementId;
    }

    //public User?: User = null; 
}

File: ControlAssuranceAPI/Models/GoElementEvidence.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class GoElementEvidence
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public bool? IsLink { get; set; }

    public string? Details { get; set; }

    public string? Controls { get; set; }

    public string? Team { get; set; }

    public string? InfoHolder { get; set; }

    public string? AdditionalNotes { get; set; }

    public int? GoElementId { get; set; }

    public DateTime? DateUploaded { get; set; }

    public int? UploadedByUserId { get; set; }

    public virtual GoElement? GoElement { get; set; }

    public virtual User? User { get; set; }
}
